From 72781109e5dadafe55f10d72ae2b3505bf0ccb93 Mon Sep 17 00:00:00 2001
From: Donald Sharp <sharpd@nvidia.com>
Date: Mon, 8 Apr 2024 16:24:05 -0400
Subject: [PATCH 10/11] zebra: Actually display I/O buffer sizes

An operator found a situation where zebra was
backing up in a significant way towards BGP
with EVPN changes taking up some serious amounts
of memory.  The key lines that would have clued
us in on it were behind a dev build.  Let's change
this.

Signed-off-by: Donald Sharp <sharpd@nvidia.com>

diff --git a/lib/stream.h b/lib/stream.h
index a3c148c9c9..7acbbd2dc7 100644
--- a/lib/stream.h
+++ b/lib/stream.h
@@ -120,9 +120,7 @@ struct stream_fifo {
 
 	/* number of streams in this fifo */
 	atomic_size_t count;
-#if defined DEV_BUILD
 	atomic_size_t max_count;
-#endif
 
 	struct stream *head;
 	struct stream *tail;
diff --git a/zebra/zserv.c b/zebra/zserv.c
index de6e404fc4..daccfa59d5 100644
--- a/zebra/zserv.c
+++ b/zebra/zserv.c
@@ -1130,12 +1130,10 @@ static void zebra_show_client_detail(struct vty *vty, struct zserv *client)
 	vty_out(vty, "ES-EVI      %-12u%-12u%-12u\n",
 		client->local_es_evi_add_cnt, 0, client->local_es_evi_del_cnt);
 	vty_out(vty, "Errors: %u\n", client->error_cnt);
-
-#if defined DEV_BUILD
 	vty_out(vty, "Input Fifo: %zu:%zu Output Fifo: %zu:%zu\n",
 		client->ibuf_fifo->count, client->ibuf_fifo->max_count,
 		client->obuf_fifo->count, client->obuf_fifo->max_count);
-#endif
+
 	vty_out(vty, "\n");
 }
 
-- 
2.17.1

